editoast/src/views/timetable/occupancy_block.rs

```diff
@@ -0,0 +1,189 @@
+use editoast_models::DbConnection;
+use editoast_schemas::primitives::Identifier;
+use itertools::izip;
+use serde::Deserialize;
+use serde::Serialize;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::sync::Arc;
+use tracing::info;
+use utoipa::ToSchema;
+
+use crate::ValkeyClient;
+use crate::ValkeyConnection;
+use crate::core::AsCoreRequest;
+use crate::core::CoreClient;
+use crate::core::pathfinding::TrackRange;
+use crate::core::signal_projection::SignalUpdate;
+use crate::core::signal_projection::SignalUpdatesRequest;
+use crate::core::signal_projection::TrainSimulation;
+use crate::error::Result;
+use crate::models;
+use crate::models::infra::Infra;
+use crate::models::prelude::*;
+use crate::views::projection::ProjectPathInput;
+use crate::views::projection::TrainSimulationDetails;
+use crate::views::projection::extract_train_details;
+use crate::views::timetable::simulation::train_simulation_batch;
+use crate::views::timetable::train_schedule::TrainScheduleError;
+
+/// Occupancy block output is described by time-space points and blocks
+#[derive(Debug, Clone, Deserialize, Serialize, ToSchema)]
+pub struct OccupancyBlockResult {
+    /// list of signal updates along the path
+    #[schema(inline)]
+    pub signal_updates: Vec<SignalUpdate>,
+}
+
+#[derive(Debug, Deserialize, ToSchema)]
+pub struct OccupancyBlockForm {
+    pub infra_id: i64,
+    pub electrical_profile_set_id: Option<i64>,
+    pub ids: HashSet<i64>,
+    #[schema(inline)]
+    pub path: ProjectPathInput,
+}
+
+/// Compute the signal updates of a list of train schedules
+pub async fn compute_batch_signal_updates<'a>(
+    core: Arc<CoreClient>,
+    infra: &Infra,
+    path_track_ranges: &'a Vec<TrackRange>,
+    path_routes: &'a Vec<Identifier>,
+    path_blocks: &'a Vec<Identifier>,
```

Comment:
```suggestion
    path_track_ranges: &'a [TrackRange]
```

I see no reason not to use a slice here and in `core::SignalUpdatesRequest`.  Could you try to change that in another commit or PR?

Reply:
done